### PR TITLE
Working on #2972 Template Conversion to Twig Format (usercp_avatar)

### PR DESCRIPTION
**Pull request by Shade-**
For #2972. Templates converted:
- [x] usercp_avatar
- [x] usercp_avatar_auto_resize_auto
- [x] usercp_avatar_auto_resize_user
- [x] usercp_avatar_current
- [x] usercp_avatar_remote
- [x] usercp_avatar_remove
- [x] usercp_avatar_upload

Templates not converted, related to avatar and used in other templates:
- usercp_currentavatar

This PR also deletes `usercp_avatar_intermediate` hook.

### CODE REVIEW

File: inc/views/base/usercp/avatar.twig [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                            <tr>
+                                <td>
+                                    {{ lang.avatar_note }}
+                                    {% for note in extranotes %}
+                                        <br />{{ note }}
+                                    {% endfor %}
+                                </td>
+                                <td class="trow1" valign="top" align="center" width="1">
+                                    <div class="usercp_container">
+                                        <img src="{{ useravatar.image }}" alt="{{ mybb.user.username|e }}" title="{{ mybb.user.username|e }}" {{ useravatar.width_height|raw }} />
```
Comment (euantorano) [inline inc/views/base/usercp/avatar.twig]: Is the `|e` filter needed here? Everything should be escaped by default.

File: usercp.php [hunk 2]
```diff
@@ -1462,14 +1462,14 @@
         );
         $db->update_query("users", $updated_avatar, "uid='".$mybb->user['uid']."'");
         remove_avatars($mybb->user['uid']);
-    } elseif($_FILES['avatarupload']['name']) // upload avatar
+    } else if ($_FILES['avatarupload']['name']) // upload avatar
```
Comment (euantorano) [inline usercp.php]: Since we've changed the convention here, can we also move the `{` to the same line?

File: usercp.php [hunk 3]
```diff
@@ -1481,7 +1481,7 @@
             );
             $db->update_query("users", $updated_avatar, "uid='".$mybb->user['uid']."'");
         }
-    } elseif($mybb->settings['allowremoteavatars']) // remote avatar
+    } else if ($mybb->settings['allowremoteavatars']) // remote avatar
```
Comment (euantorano) [inline usercp.php]: And again, code style.

**Comment by Shade-** [comment]
Fixed.

**Pull request merged**